ui/jobs/bar.d.ts

```diff
@@ -0,0 +1,61 @@
+import { Job } from '../../types/job';
+import { JobDetail } from '../../types/event';
+import { ComboCallback } from './combo_tracker';
+
+export interface TimerBar {
+  duration: number;
+}
+
+export interface Box extends HTMLElement {
+  elapsed: number;
+  duration: number;
+  threshold: number;
+  valuescale: number;
+  fg: string;
+  // parentNode: { classList: ()};
+}
+
+type Selector = {
+  id?: string;
+  fgColor?: string;
+  classList?: string[];
+  threshold?: number;
+};
+
+interface ResourceBox extends HTMLDivElement {
+  parentNode: {
+    classList: {
+      add: (s: string) => void;
+      remove: (s: string) => void;
+    };
```

Comment:
It's a little weird to redefine types from `HTMLElement` and to also assert that a particular piece of HTML is what we think it is (which it might not be)?  What about just adding a two liner to verify that the type is what we expect it to be, and then TypeScript can infer that `p` is an `HTMLElement` with `classList` already?

```typescript
    const p = bloodBox.parentNode;
    if (!p || !(p instanceof HTMLElement))
      return;
```

Reply:
`parentNode: HTMLElemen` should be fine enough?